libs/common/src/change-stream/delete-many.ts

```diff
@@ -68,17 +84,20 @@ const deleteManyWithAuditLogs = async ({ filter, collectionName, db, auditDetail
  * @throws {DocumentNotFoundError} - if there are no documents matching the filter to delete
  * @throws {WriteConcernError} - if either the deletion-audit-log insertion or document deletion operations are not acknowledged
  */
-export const deleteMany = async ({ filter, collectionName, db, auditDetails }: DeleteManyParams): Promise<{ acknowledged: boolean }> => {
+export const deleteMany = async <CollectionName extends MongoDbCollectionName>({
+  filter,
+  collectionName,
+  db,
+  auditDetails,
+}: DeleteManyParams<CollectionName>): Promise<DeleteResult> => {
   if (process.env.CHANGE_STREAM_ENABLED === 'true') {
-    await deleteManyWithAuditLogs({
+    return await deleteManyWithAuditLogs({
       filter,
       collectionName,
       db,
       auditDetails,
     });
-
-    return { acknowledged: true };
   }
   const collection = await db.getCollection(collectionName);
-  return collection.deleteMany(filter);
+  return await collection.deleteMany(filter);
```

Comment:
is this await (and the one above) redundant as we're not inside a try catch

Reply:
cool, am happy with that, I tend to lean the other way of "its not needed so lets leave it out", but happy for this